src/gateway/WebSocket.ts

```diff
@@ -3,39 +3,52 @@ import WebSocket from 'ws';
 import { EVENTS, Payload } from './constants';
 import { GATEWAY_OPCODES, GATEWAY_CLOSE_EVENT_CODES } from '../util/Constants';
 
-export default function Socket(token: string, wsURL: string, intents: number) {
-  let interval: NodeJS.Timeout;
-  let lastHeartbeatAck = true;
-  let session_id: string;
-  let seq: number;
-  let ws: WebSocket;
-
-  function heartbeat(expected: boolean) {
+export default class Socket {
+  private _intents: number;
+  private _token: string;
+  private _wsURL: string;
+  private interval!: NodeJS.Timeout;
+  private lastHeartbeatAck = true;
+  private seq!: number;
+  private session_id!: string;
+  private ws!: WebSocket;
+
+  constructor(token: string, wsURL: string, intents: number) {
+    this._token = token;
+    this._wsURL = wsURL;
+    this._intents = intents;
+
+    this.identify = this.identify.bind(this);
```

Comment:
While this shouldn't be an issue at runtime, I'd probably use a different naming for the bound function? @bsian03 though